server/neptune/temporalworker/job/store.go

```diff
@@ -171,6 +166,12 @@ func (s StorageBackendJobStore) Write(ctx context.Context, jobID string, output
 
 // Activity context since it's called from within an activity
 func (s *StorageBackendJobStore) Close(ctx context.Context, jobID string, status JobStatus) error {
+	// Since we close the job in a different activity than when it's created, it is possible that we try closing a non existent job
+	// after the worker has been restarted. So, instead of hard failing, let's return since we close all jobs in progress during shutdown
+	if job, err := s.InMemoryStore.Get(ctx, jobID); job == nil && err == nil {
+		return nil
+	}
```

Comment:
Oh shit, if this is the problem, then this needs to be rethought.  Even with the shutdown solution I think we'd still run into this since shutting down a pod with an in-progress activity wouldn't actually move on to the close job activity since the worker would stop taking more tasks.

This implementation is effectively broken now if there's a deploy during plan time or apply time.  We probably need to come up with a backup plan here.  

I think maybe what we should do is just move Terraform init and Terraform plan into one activity and call close job from within that at the end.  Same with Terraform Apply. We can talk about it as a team but I don't see another way around it.

Reply:
As discussed offline, this is fine because we persist during shutdown anyways.